code/game/objects/structures/vehicles/secway.dm

```diff
@@ -58,3 +59,228 @@
 	//icon_state = "gokart_wreck"
 	name = "secway wreckage"
 	desc = "Nothing to see here!"
+
+var/list/random_tool_sounds = list('sound/items/Ratchet.ogg','sound/items/Screwdriver.ogg', 'sound/items/Screwdriver2.ogg',
+	'sound/items/Wirecutter.ogg', 'sound/weapons/toolhit.ogg','sound/items/Welder.ogg', 'sound/items/Welder2.ogg',
+	'sound/items/Crowbar.ogg')
+
+var/list/descriptive_sprites = list("I go for the classics", "A big donut", "A Rottweiler combat cyborg", "I'm the head honcho", "A valkyrie's chariot")
+
+/obj/item/weapon/secway_kit
+	name = "custom secway kit"
+	desc = "Everything you need to build your own custom Secway."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "modkit"
+	flags = FPRINT
+	w_class = W_CLASS_SMALL
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/newsprites_lefthand.dmi', "right_hand" = 'icons/mob/in-hand/right/newsprites_righthand.dmi')
+	var/remaining_upgrades = 3
+	var/obj/structure/bed/chair/vehicle/secway/custom/baby
+	var/named = FALSE
+
+/obj/item/weapon/secway_kit/New()
+	..()
+	baby = new(src)
+
+/obj/item/weapon/secway_kit/examine(mob/user)
+	..()
+	if(!named)
+		to_chat(user,"<span class='warning'>It needs a name!</span>")
+	to_chat(user,"<span class='warning'>It has room for [remaining_upgrades] more upgrades.</span>")
+	if(remaining_upgrades < 1)
+		return
+	if(!baby.can_have_carts)
+		to_chat(user,"<span class='info'>There's space for a cart hook made of metal sheeting.</span>")
+	if(baby.max_health < 300)
+		to_chat(user,"<span class='info'>It hasn't been upgraded with plasteel for armoring.</span>")
+	if(!(baby.pass_flags & PASSMOB))
+		to_chat(user,"<span class='info'>If you rubbed ectoplasm on it, it would cruise through people instead of bumping them.</span>")
+	if(!is_type_in_list(/datum/action/vehicle/toggle_headlights,baby.vehicle_actions))
+		to_chat(user,"<span class='info'>Noone has added a flashlight for headlights or a siren helmet for police lights.</span>")
+	if(baby.knockdown_time < 3)
+		to_chat(user,"<span class='info'>A sheet of plastic would make a good bumper for knocking people over.</span>")
+	if(baby.hit_damage < 5)
+		to_chat(user,"<span class='info'>You could probably stick on a knife to jab people you run into.</span>")
+	if(baby.can_take_pai == FALSE)
+		to_chat(user,"<span class='info'>With an unprinted circuitboard you could slot in a personal AI.</span>")
+	if(!baby.impact_sound)
+		to_chat(user,"<span class='info'>There's a good spot for a hailer to make legal noises when you hit stuff.</span>")
+	if(!baby.can_spacemove)
+		to_chat(user,"<span class='info'>A spacepod core is just what this thing needs to move in space.</span>")
+	if(baby.movement_delay > 0.9)
+		to_chat(user,"<span class='info'>If it had an RTG cell, this thing could go a little faster.</span>")
+	if(baby.explodes)
+		to_chat(user,"<span class='info'>With an energy gun, it could be rigged to fire a countermeasure if destroyed up.</span>")
```

Comment:
The grammar at the end is a little confusing. Does it proc when it's destroyed or shot up?

Reply:
When destroyed, it's a typo

> imagine writing feedback messages at 3am

Reply:
I can't sleep and I saw an interesting new PR. Help.

Reply:
I actually meant myself, the line you (correctly) pointed out was wrong is a feedback message.

We've both lost control of our lives